### PR TITLE
Ali pom

### PR DESCRIPTION
**Pull request by fairwinds0099**

### CODE REVIEW

File: maven-restassured-jdbc.iml [hunk 1]
```diff
@@ -5,32 +5,120 @@
     <output-test url="file://$MODULE_DIR$/target/test-classes" />
     <content url="file://$MODULE_DIR$">
       <sourceFolder url="file://$MODULE_DIR$/src/main/java" isTestSource="false" />
-      <sourceFolder url="file://$MODULE_DIR$/src/test/java" isTestSource="true" />
+      <sourceFolder url="file://$MODULE_DIR$/src/main/resources" isTestSource="false" />
```
[the remark this answers is not in the document]
Comment (jacobbrendall) [reply maven-restassured-jdbc.iml]: sure

File: pom.xml [hunk 2]
```diff
@@ -103,6 +99,21 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-java</artifactId>
+            <version>3.11.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-server</artifactId>
+            <version>3.11.0</version>
```
[the remark this answers is not in the document]
Comment (jacobbrendall) [reply pom.xml]: deleted the replicates

File: src/main/java/domain/pages/DirectoryPage.java [hunk 3]
```diff
@@ -1,22 +1,55 @@
...
+
+public class DirectoryPage {
+    private WebDriver driver;
+    public String stateTextXpath = "//*[@id='%s']/..//td[2]";
+    public final String partyXpath = "(//a[contains(text(),'Byrne, Bradley')])[1]/../following-sibling::td[1]";
+    public List<WebElement> states;
+
+
+    @FindBy(xpath =  "//h1[contains (text(), 'Directory of Representatives')]")
+    private WebElement directoryOfRepresentativesText;
```
[the remark this answers is not in the document]
Comment (jacobbrendall) [reply src/main/java/domain/pages/DirectoryPage.java]: done

File: src/main/java/domain/pages/DirectoryPage.java [hunk 4]
```diff
@@ -1,22 +1,55 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DirectoryPage {
+    private WebDriver driver;
+    public String stateTextXpath = "//*[@id='%s']/..//td[2]";
+    public final String partyXpath = "(//a[contains(text(),'Byrne, Bradley')])[1]/../following-sibling::td[1]";
```
[the remark this answers is not in the document]
Comment (jacobbrendall) [reply src/main/java/domain/pages/DirectoryPage.java]: done

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (fairwinds0099) [reply pom.xml]: do we need sleenium- java anad selenium-server at the same time? what happens if we remove

File: maven-restassured-jdbc.iml (hunk 1 above)
[the remark this answers is not in the document]
Comment (fairwinds0099) [reply maven-restassured-jdbc.iml]: I don't know why there are tons of chnages each time:(

File: src/main/java/tests/enterZipCodeTest.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import domain.pages.HomePage;
+import org.junit.Test;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.PageFactory;
+
+public class enterZipCodeTest {
+
+    @Test
+            public void isZipCodeCorrect()
```
Comment (fairwinds0099) [inline src/main/java/tests/enterZipCodeTest.java]: Please add a description of what the test does
update method name accordingly...

File: src/main/java/tests/enterZipCodeTest.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    System.()
```
Comment (fairwinds0099) [inline src/main/java/tests/enterZipCodeTest.java]: System.()?

File: src/main/java/tests/UiTestTemp.java [hunk 7]
```diff
@@ -11,30 +11,30 @@
...
-
-        driver.get("https://www.house.gov");
-        homePage.clickLeaderShipLink();
-
-        Assert.assertTrue(leadershipPage.isLeadershipTitlePresent());
-    }
+        //System.setProperty("webdriver.chrome.driver", "resources/chromedriver");
+        //WebDriver driver = new ChromeDriver();
+        //HomePage homePage = new PageFactory().initElements(driver, HomePage.class);
+
```
Comment (fairwinds0099) [inline src/main/java/tests/UiTestTemp.java]: do not leave any comments in your code please

File: src/main/java/tests/UiTest.java [hunk 8]
```diff
@@ -2,15 +2,32 @@
...
+import org.openqa.selenium.support.PageFactory;
+
 
 public class UiTest {
 
-    WebDriver driver;
 
-    DirectoryPage directoryPage = PageFactory.initElements(driver, DirectoryPage.class);
+    @Test
+    public void listByStateandDistrict() {
```
Comment (fairwinds0099) [inline src/main/java/tests/UiTest.java]: rename method include should into the method name

File: src/main/java/tests/GetPartyByRep.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
+package tests;
+
+import domain.pages.DirectoryPage;
+import domain.pages.HomePage;
+import org.junit.Test;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.PageFactory;
+
+public class GetPartyByRep {
```
Comment (fairwinds0099) [inline src/main/java/tests/GetPartyByRep.java]: rename class PartyNameTests

File: src/main/java/tests/GetPartyByRep.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+    @Test
+    public void getPartyByReps(){
+        System.setProperty("webdriver.chrome.driver", "resources/chromedriver");
+        WebDriver driver = new ChromeDriver();
+        driver.get("https://www.house.gov");
+        HomePage hp = new PageFactory().initElements(driver, HomePage.class);
+        hp.clickRepresentativesLink();
+        DirectoryPage dp = new PageFactory().initElements(driver, DirectoryPage.class);
```
Comment (fairwinds0099) [inline src/main/java/tests/GetPartyByRep.java]: directoryPage

File: src/main/java/tests/GetPartyByRep.java [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+        dp.isDisplayedDirectoryOfRepresentativesText();
+        dp.clickListByStateandDistrict();
+        dp.isAlabamaTextDisplayed();
+        System.out.println(dp.getPartyByRepresentative("Grijalva"));
```
Comment (fairwinds0099) [inline src/main/java/tests/GetPartyByRep.java]: use Assertions

File: src/main/java/tests/GetAllStates.java [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+import domain.pages.HomePage;
+import org.junit.Test;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.PageFactory;
+
+public class GetAllStates {
+
+    @Test
+    public void getAllStatesofReps() {
```
Comment (fairwinds0099) [inline src/main/java/tests/GetAllStates.java]: Of

File: src/main/java/tests/GetAllRepresentativesNamesForSelected.java [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+    @Test
+    public void getAllRepsNamesForSelected(){
+        System.setProperty("webdriver.chrome.driver", "resources/chromedriver");
+        WebDriver driver = new ChromeDriver();
+        driver.get("https://www.house.gov");
+        HomePage hp = new PageFactory().initElements(driver, HomePage.class);
+        hp.clickRepresentativesLink();
+        DirectoryPage dp = new PageFactory().initElements(driver, DirectoryPage.class);
+        dp.clickListByStateandDistrict();
+        System.out.println(dp.getAllRepresentativeNamesForSelectedState("state-alabama").toString());
```
Comment (fairwinds0099) [inline src/main/java/tests/GetAllRepresentativesNamesForSelected.java]: remove souts and add assertions

File: src/main/java/tests/GetAllRepresentativesNamesForSelected.java [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.PageFactory;
+
+public class GetAllRepresentativesNamesForSelected {
+    @Test
+    public void getAllRepsNamesForSelected(){
+        System.setProperty("webdriver.chrome.driver", "resources/chromedriver");
+        WebDriver driver = new ChromeDriver();
+        driver.get("https://www.house.gov");
+        HomePage hp = new PageFactory().initElements(driver, HomePage.class);
```
[the remark this answers is not in the document]
Comment (fairwinds0099) [reply src/main/java/tests/GetAllRepresentativesNamesForSelected.java]: use homePAge as object name

File: src/main/java/domain/pages/HomePage.java [hunk 15]
```diff
@@ -1,4 +1,20 @@
...
 public class HomePage {
+
+    @FindBy (xpath = "(//a[@href = '/representatives'])[1]")
+    private WebElement representativesLink;
+
+    public void clickRepresentativesLink(){
+        representativesLink.click();
+    }
+
+    public void enterZipCode(String zipCode){}
```
[the remark this answers is not in the document]
Comment (jacobbrendall) [reply src/main/java/domain/pages/HomePage.java]: done

File: src/main/java/domain/pages/DirectoryPage.java [hunk 16]
```diff
@@ -1,22 +1,55 @@
...
+    public List<WebElement> states;
+
+
+    @FindBy(xpath =  "//h1[contains (text(), 'Directory of Representatives')]")
+    private WebElement directoryOfRepresentativesText;
+    @FindBy (linkText = "By State and District")
+    private WebElement listByStateandDistrict;
```
[the remark this answers is not in the document]
Comment (jacobbrendall) [reply src/main/java/domain/pages/DirectoryPage.java]: changed to listByStateandDistrictButton

File: src/main/java/domain/pages/HomePage.java [hunk 17]
```diff
@@ -1,4 +1,20 @@
 package domain.pages;
 
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
 public class HomePage {
+
+    @FindBy (xpath = "(//a[@href = '/representatives'])[1]")
```
[the remark this answers is not in the document]
Comment (jacobbrendall) [reply src/main/java/domain/pages/HomePage.java]: thanks

File: src/main/java/tests/enterZipCodeTest.java (hunk 6 above)
Comment (jacobbrendall) [reply src/main/java/tests/enterZipCodeTest.java]: done

File: src/main/java/tests/UiTestTemp.java (hunk 7 above)
Comment (jacobbrendall) [reply src/main/java/tests/UiTestTemp.java]: got it

File: src/main/java/tests/UiTest.java (hunk 8 above)
Comment (jacobbrendall) [reply src/main/java/tests/UiTest.java]: done

File: src/main/java/tests/GetPartyByRep.java (hunk 9 above)
Comment (jacobbrendall) [reply src/main/java/tests/GetPartyByRep.java]: done

File: src/main/java/tests/GetPartyByRep.java (hunk 11 above)
Comment (jacobbrendall) [reply src/main/java/tests/GetPartyByRep.java]: done

File: src/main/java/tests/GetAllStates.java (hunk 12 above)
Comment (jacobbrendall) [reply src/main/java/tests/GetAllStates.java]: done

File: src/main/java/tests/GetAllRepresentativesNamesForSelected.java (hunk 13 above)
Comment (jacobbrendall) [reply src/main/java/tests/GetAllRepresentativesNamesForSelected.java]: done

File: src/main/java/tests/UiTestTemp.java (hunk 7 above)
Comment (jacobbrendall) [reply src/main/java/tests/UiTestTemp.java]: done

File: src/main/java/tests/GetAllRepresentativesNamesForSelected.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (jacobbrendall) [reply src/main/java/tests/GetAllRepresentativesNamesForSelected.java]: done

File: src/main/java/tests/GetAllRepresentativesNamesForSelected.java [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+        hp.clickRepresentativesLink();
+        DirectoryPage dp = new PageFactory().initElements(driver, DirectoryPage.class);
```
[the remark this answers is not in the document]
Comment (jacobbrendall) [reply src/main/java/tests/GetAllRepresentativesNamesForSelected.java]: done

File: src/main/java/tests/GetAllRepresentativesNamesForSelected.java (hunk 14 above)
[the remark this answers is not in the document]
Comment (jacobbrendall) [reply src/main/java/tests/GetAllRepresentativesNamesForSelected.java]: done

**Pull request merged**